src/main/java/org/isf/dicom/manager/SourceFiles.java

```diff
@@ -134,12 +134,17 @@ private void loadDicomDir(FileDicom fileDicom, File sourceFile, int patient) thr
 				}
 				filesLoaded++;
 				dicomLoader.setLoaded(filesLoaded);
-			} else if (!".".equals(value.getName()) && !"..".equals(value.getName())) {
+			} else if (isValidDirectoryName(value.getName())) {
 				loadDicomDir(fileDicom, value, patient);
 			}
 		}
 	}
 
+	private static boolean isValidDirectoryName(String directoryName) {
+		return ".".equals(directoryName) && "..".equals(directoryName);
```

Comment:
This change also introduced a new bug; the code should be 
```suggestion
		return !".".equals(directoryName) && !"..".equals(directoryName);
```